SDL2: fix cross-compilation

Mindavi opened the pull request:
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux (cross-compiled)
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Mindavi committed:
SDL2: fix cross-compilation

Mindavi commented:
Maybe doing this would be sufficient?

https://github.com/Mindavi/nixpkgs/commit/faf6a266badf04e23aed926fdc2fbf13e7005bee

NickCao commented:
We can do better by switching to cmake, https://github.com/NickCao/nixpkgs/commit/b4c6b4e21360b0129d180700d8d3dcf69eaa00a6, it alse greatly simplifies the packaging.

ehmry commented:
I think it would be better to put wayland-scanner in `nativeBuildInputs` and replace `$(WAYLAND_SCANNER)` with simply `wayland-scanner`. Taking `buildPackages` as function argument makes it difficult to override waylang-scanner externally.

Mindavi committed:
SDL2: use wayland-scanner from PATH

Prevents having to use buildPackages, helps overriding wayland-scanner for whoever needs it.

Mindavi commented:
Updated based on the suggestion by @ehmry. I think this makes sense. Building now... Not expecting any issues with this alternative approach (and it looks cleaner too).
